juju/juju
# Remove assumption that Azure security group is hard coded by Juju

## Description of change

When specifying a bespoke virtual network for a Juju model on azure, use the security group (if any) associated with the user specified subnets instead of Juju creating one with a hard coded name.
Juju will now only create network resources - virtual networks, subnets, security groups - for models where there is no user specified network model-config.
When opening/closing ports on an instance, Juju will look up the security groups to use off the instances's NIC(s) instead of assuming a hard coded one. Followup work needed to consider the new endpoint binding modelling.

## QA steps

bootstrap on azure
juju add-model and ensure deploy mysql and expose work as before with the hard coded artefacts
create a bespoke resource group (test-rg) and virtual network (test-vn) with subnets and a security group
juju add-model --config network=test-rg/test-vn
juju add-space to create a space with one of the subnets
juju deploy mysql --bind db=myspace --constraints="spaces=alpha"
juju expose mysql
check that the model's resource group has the machine and NICs, and that these use the subnets in the bespoke vnet
check the bespoke security group for the mysql port rule
juju show-unit mysql/0 and check the network info matches the bound space subnet
